app/connectors/referenceData/GetMemberStatesHttpParser.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.referenceData
+
+import connectors.BaseConnectorUtils
+import models.MemberStateModel
+import models.response.{ErrorResponse, JsonValidationError, UnexpectedDownstreamResponseError}
+import play.api.http.Status.OK
+import play.api.libs.json.Reads
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpReads, HttpResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait GetMemberStatesHttpParser extends BaseConnectorUtils[Seq[MemberStateModel]] {
+
+  implicit val reads: Reads[Seq[MemberStateModel]] = Reads.seq(MemberStateModel.format.reads)
+  def http: HttpClient
+
+  class GetMemberStatesReads() extends HttpReads[Either[ErrorResponse, Seq[MemberStateModel]]] {
+    override def read(method: String, url: String, response: HttpResponse): Either[ErrorResponse, Seq[MemberStateModel]] = {
+      response.status match {
+        case OK =>
+          response.validateJson match {
+            case Some(valid: Seq[MemberStateModel]) if valid.isDefinedAt(0) => Right(valid)
+            case None =>
```

Comment:
This should be an underscore, as there's a case missing. If it's Some(_) and then empty would throw an exception.